### PR TITLE
Attempt to fix "Gave up!" problem in prop_balanceTransactionUnresolvedInputs

### PR DESCRIPTION
**Pull request by Anviking**
- Simply removing `==>` seems to work fine, and I believe makes it more likely to
  pass.
- Remove `withMaxSuccess` because it seems to be doing nothing in
  conjunction with `checkCoverage`.

### Comments

- I'm not entirely sure of the interplay between the various `==>`, `checkCoverage`, etc, not certain that this _for sure_ is a fix, however, it can't hurt to try.

```
$ cabal run cardano-wallet:test:unit -- --match "unresolved inputs" --seed 1053383965                                                                                                                                                    
Cardano.Wallet.Shelley.Transaction
  balanceTransaction
    when passed unresolved inputs
      may fail (1697ms)
        +++ OK, passed 100 tests:
        10% unknown txins

        87% other error
         3% nothing dropped

Finished in 1.6980 seconds, used 1.7075 seconds of CPU time
1 example, 0 failures
```

### Issue Number

ADP-1590

### CODE REVIEW

File: lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs [hunk 1]
```diff
@@ -2351,22 +2351,26 @@ prop_balanceTransactionUnresolvedInputs
...
-                           -- Balancing can succeed if the dropped inputs happen
-                           -- to be apart of the wallet UTxO.
+                        | null dropped
+                            -> label "nothing dropped"
+                                $ property True
+                        | otherwise
+                            -> label "succeeded despite unresolved input"
+                                $ property True
+                            -- Balancing can succeed if the dropped inputs happen
+                            -- to be apart of the wallet UTxO.
```
Comment (jonathanknowles) [inline lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs]: Hi @Anviking. Did you mean "a part of"?

```suggestion
                            -- Balancing can succeed if the dropped inputs
                            -- happen to be a part of the wallet UTxO.
```

**Review by jonathanknowles — APPROVED** [review]
Looks okay to me!

I only have one suggestion: it looks like one of the comments has a typo: I've provided a suggested fix. (I suspect it's not correct, but feel free to point out if I'm mistaken!)

File: lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs (hunk 1 above)
Comment (Anviking) [reply lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs]: Ah! Yes!

**Pull request merged**